src/runtime/testing/preconditions.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/**
+ * @fileoverview
+ * Preconditions
```

Comment:
I was imagining (following some google practices) to use `/testing/` directory for fake/stub implementations that are to be used in testing, but are not tests themselves. This will matter if/once we reach a build system which will be capable to enforce rules on deps (e.g. no prod code should depend on tests of testing, runtime should not depend on devtools etc.). I would propose to move this file to either src/runtime, or alternatively, to create src/common and have it there (together with hot.ts, and probably some other stuff?)

WDYT?

Reply:
Yes, let's discuss.  After all I did propose creating js guava!